src/main/js/main_html5.js

```diff
@@ -889,20 +890,25 @@ require("../../../html5-common/js/utils/environment.js");
      * @public
      * @method OoyalaVideoWrapper#setAudio
      * @param {String} trackId - the ID of the audio track to activate
-     * @returns {Boolean} true - if new audio track was set; false is in otherwise;
+     * @returns {Boolean} true - if new audio track was set; false otherwise;
      */
     this.setAudio = function(trackId) {
-      var audioTracks = _video.audioTracks;
-      if (audioTracks && audioTracks.length) { //if audioTracks exist
-        var newAudioTrack = audioTracks.getTrackById(trackId);
-        if (typeof newAudioTrack !== 'undefined') { //if trackId is correct and the audio exists
-          var prevAudioTrack = audioTracks.getTrackById(currentAudioId);
-          if (typeof newAudioTrack !== 'undefined') { //if currentAudioId is correct and the audio exists
-            prevAudioTrack.enabled = false; //the audio is not active anymore
+      if (this.currentAudioId !== trackId) {
+        var audioTracks = _video.audioTracks;
+        if (audioTracks && audioTracks.length) { //if audioTracks exist
+
+          var newAudioTrack = audioTracks.getTrackById(trackId);
+          if (newAudioTrack && typeof newAudioTrack !== 'undefined') { //if trackId is correct and the audio exists
```

Comment:
I think we could do a simpler check here:
``` if (newAudioTrack)```
`newAudioTrack` will be falsy if it's undefined, so the second check might be redundant. Same comment for the other checks in this function.

Reply:
\+